libraries/chain/controller.cpp

```diff
@@ -3993,80 +3992,78 @@ struct controller_impl {
       auto qc_ext_id = quorum_certificate_extension::extension_id();
       bool qc_extension_present = block_exts.count(qc_ext_id) != 0;
 
-      EOS_ASSERT( !qc_extension_present, invalid_qc_claim,
+      EOS_ASSERT( !qc_extension_present, block_validate_exception,
                   "Legacy block #${b} includes a QC block extension",
                   ("b", block_num) );
 
-      EOS_ASSERT( !b->is_proper_svnn_block(), invalid_qc_claim,
+      EOS_ASSERT( !b->is_proper_svnn_block(), block_validate_exception,
                   "Legacy block #${b} has invalid schedule_version",
                   ("b", block_num) );
 
       // Verify we don't go back from Savanna (Transition or Proper) block to Legacy block
-      auto it = prev.header_exts.find(finality_extension::extension_id());
-      EOS_ASSERT( it == prev.header_exts.end(), invalid_qc_claim,
-                  "Legacy block #${b} has previous block which contains finality block header extension",
-                  ("b", block_num) );
-      EOS_ASSERT( !prev.header.is_proper_svnn_block(), invalid_qc_claim,
-                  "Legacy block #${b} may not have previous block that is a Proper Savanna block",
+      EOS_ASSERT( prev.header.is_legacy_block(), block_validate_exception,
+                  "Legacy block #${b} must have previous block that is also a Legacy block",
                   ("b", block_num) );
    }
 
    void verify_transition_block_exts( const signed_block_ptr& b, const block_header_state_legacy& prev ) {
-      assert(!b->is_legacy_block());
+      assert(!b->is_legacy_block() && !b->is_proper_svnn_block());
 
       uint32_t block_num = b->block_num();
       auto block_exts = b->validate_and_extract_extensions();
       auto qc_ext_id = quorum_certificate_extension::extension_id();
       bool qc_extension_present = block_exts.count(qc_ext_id) != 0;
 
-      EOS_ASSERT( !qc_extension_present, invalid_qc_claim,
+      EOS_ASSERT( !qc_extension_present, block_validate_exception,
```

Comment:
For consistency, this should either remain `invalid_qc_claim` or we should change similar checks involving the integrity of the QC block extension in `verify_proper_block_exts`.